frontend/src/Discover/Map.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useEffect, useRef } from 'react';
+import L from 'leaflet';
+import axios from 'axios';
+import 'leaflet/dist/leaflet.css';
+import './Map.css';
+
+export default function Map() {
+  const mapRef = useRef(null);
+  const legendRef = useRef(null);
+
+  useEffect(() => {
+    if (mapRef.current) return;
+
+    const maxBounds = [
+      [-85, -180],
+      [85, 180]
+    ];
+
+    mapRef.current = L.map('map', {
+      center: [20, 0], // Center the map to display all continents
+      zoom: 3,
+      minZoom: 3,      // Minimum zoom level
+      maxZoom: 8,      // Maximum zoom level
+      maxBounds: maxBounds,
+      maxBoundsViscosity: 1.0 // Ensures that the map cannot be zoomed out beyond the bounds
+    });
+
+    mapRef.current.on('drag', function () {
+      mapRef.current.panInsideBounds(maxBounds, { animate: false });
+    });
+
+    L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+    }).addTo(mapRef.current);
+
+    const fetchGeoJSON = async () => {
+      const response = await fetch('/countries.geojson');
+      return response.json();
+    };
+
+    const fetchData = async () => {
+      try {
+        const response = await axios.get('https://disease.sh/v3/covid-19/vaccine/coverage/countries?lastdays=1');
+        const data = response.data;
+
+        const vaccineData = {};
```

Comment:
could we leave some more high-level comments on what the different pieces of logic are? to make the code a bit more understandable